proto/v1/vulnerability.proto

```diff
@@ -41,54 +41,61 @@ enum Severity {
   CRITICAL = 5;
 }
 
-// Vulnerability provides metadata about a security vulnerability in a Note.
+// A security vulnerability that can be found in resources.
 message VulnerabilityNote {
-  // The CVSS score for this vulnerability.
+  // The CVSS score of this vulnerability. CVSS score is on a scale of 0 - 10
+  // where 0 indicates low severity and 10 indicates high severity.
   float cvss_score = 1;
 
-  // Note provider assigned impact of the vulnerability.
+  // The note provider assigned severity of this vulnerability.
   Severity severity = 2;
 
-  // All information about the package to specifically identify this
-  // vulnerability. One entry per (version range and cpe_uri) the package
-  // vulnerability has manifested in.
+  // Details of all known distros and and packages affected by this
+  // vulnerability.
   repeated Detail details = 3;
 
-  // Identifies all appearances of this vulnerability in the package for a
-  // specific distro/location. For example: glibc in
-  // cpe:/o:debian:debian_linux:8 for versions 2.1 - 2.2
+  // A detail for a distro and package affected by this vulnerability and its
+  // associated fix (if one is available).
   message Detail {
-    // Required. The CPE URI in
-    // [cpe format](https://cpe.mitre.org/specification/) in which the
-    // vulnerability manifests. Examples include distro or storage location for
-    // vulnerable jar.
-    string cpe_uri = 1;
+    // The distro assigned severity of this vulnerability.
+    string severity_name = 1;
+
+    // A vendor-specific description of this vulnerability.
+    string description = 2;
+
+    // The type of package; whether native or non native (e.g., ruby gems,
+    // node.js packages, etc.).
+    string package_type = 3;
 
-    // Required. The name of the package where the vulnerability was found.
-    string package = 2;
+    // Required. The [CPE URI](https://cpe.mitre.org/specification/) this
+    // vulnerability affects.
+    string affected_cpe_uri = 4;
 
-    // The min version of the package in which the vulnerability exists.
-    grafeas.v1.Version min_affected_version = 3;
+    // Required. The package this vulnerability affects.
+    string affected_package = 5;
 
-    // The severity (eg: distro assigned severity) for this vulnerability.
-    string severity_name = 4;
+    // Required. The minimum version of the package this vulnerability affects.
+    grafeas.v1.Version affected_version = 6;
```

Comment:
I think leaving `min_` in this name might be clearer. I could be convinced otherwise.

Reply:
Maybe this is better.

Reply:
Added min_ back and also added it to the vuln occurrence's package issue affected location field as well.